hi-ml-azure/src/health_azure/utils.py

```diff
@@ -2021,3 +2108,205 @@ def check_is_any_of(message: str, actual: Optional[str], valid: Iterable[Optiona
+    :param subscription_id: An optional subscription ID.
+    :param resource_group: An optional resource group name.
+    :param workspace_name: An optional workspace name.
+    :return: An instance of MLClient to interact with Azure resources.
+    """
+    if ml_client:
+        return ml_client
+
+    credential = get_credential()
+    if credential is None:
+        raise ValueError("Can't connect to MLClient without a valid credential")
+    if aml_workspace is not None:
+        ml_client = MLClient(
+            subscription_id=aml_workspace.subscription_id,
+            resource_group_name=aml_workspace.resource_group,
+            workspace_name=aml_workspace.name,
+            credential=credential)  # type: ignore
+    elif workspace_config_path:
+        ml_client = MLClient.from_config(
+            credential=credential,  # type: ignore
+            path=str(workspace_config_path))
+    elif subscription_id and resource_group and workspace_name:
+        ml_client = MLClient(
+            subscription_id=subscription_id,
+            resource_group_name=resource_group,
+            workspace_name=workspace_name,
+            credential=credential)  # type: ignore
+    else:
+        try:
+            workspace = get_workspace()
+            ml_client = MLClient(
+                subscription_id=workspace.subscription_id,
+                resource_group_name=workspace.resource_group,
+                workspace_name=workspace.name,
+                credential=credential)  # type: ignore
+        except ValueError as e:
+            raise ValueError(f"Couldn't connect to MLClient: {e}")
+    logging.info(f"Logged into AzureML workspace {ml_client.workspace_name}")
+    return ml_client
+
+
+def retrieve_workspace_from_client(ml_client: MLClient, workspace_name: Optional[str] = None
+                                   ) -> WorkspaceV2:
+    """
+    Get a v2 Workspace object from an MLClient object. If a workspace_name is passed, will attempt
+    to retrieve a workspace with that name. Otherweise will use the MLClient's default workspace_name
+
+    :param ml_client: An MLClient object to retrieve the Workspace from
+    :param workspace_name: An optional name of the workspace to retrieve.
+    :return: A v2 Workspace object.
+    """
+    if workspace_name is not None:
+        workspace_name = workspace_name
+    elif ml_client.workspace_name is not None:
+        workspace_name = ml_client.workspace_name
+    else:
+        workspace_name = ""
+    workspace = ml_client.workspaces.get(workspace_name)
+    return workspace
+
+
+def fetch_job(ml_client: MLClient, run_id: str) -> Job:
+    """
+    Retrive a job with a given run_id from an MLClient
+
+    :param ml_client: An MLClient object.
+    :param run_id: The id of the run to retrieve.
+    :return: An Azure ML (v2) Job object.
+    """
+    job = ml_client.jobs.get(run_id)
```

Comment:
same comment, do we need an extra func for this?

Reply:
I think it's important to have an example of how to do this, since it's not clearly explained in the Azure ML docu